Responsive design

dipbazz opened the pull request:
**This project is about building a home page about the next web.**

The features of this project are listed below:
1. The website is completely responsive with three different breakpoints as:
    - Below 768px for mobile screen
    - In between 768px and 1024px for medium screen.
    - Above 1024px for large screen.

2. Website has a header:
    - With the next web logo.
    - Top header with useful links
    - Bottom header with website menu links, with a search icon.

3. Contains the main section with:
    - Banner news article
    - Latest news
    - Funding Source
    - Other news suggestions
    - Latest deals

4. Has a footer with
    - Social share buttons, Other useful links
    - Bottom footer with copyright info.

dipbazz committed:
Added required linter files and added elements on index file

Files added for linters are: .hintrc, stylelintrc.json, linters.yml
Elements added on HTML files for:
- Top header
- bottom header
- banner highlighted articles
- latest news section
- funding section
- suggestion news articles

smcommits committed:
Finalize HTML and Add Assets

- Inputted data into html elements
- Finalized header, banner, news-suggestions, latest-articles, fundings,
  latest deals, footer.
- Added images to the assets folder.

dipbazz committed:
Created a styles for header and banner image.

- started to style with mobile first responsive

smcommits committed:
Update HTML and CSS

HTML
- Wrapped latest news heading and info in one div
- Wrapped funding-footer text

CSS
- Styled latest news using flex and image properties
- Styled funding using absolute and flex

dipbazz committed:
Updated index.html and style.css file

refactored HTML elements to place a tag.
and grouped some content where necessary.

updated css file to update new changes on html file.

smcommits committed:
Progress with HTML and CSS

HTML
- Grouped different items together belonging to new-suggestion and
  latest deals class to use flex property.

CSS
- Progressed with the style of news-suggestion element. Used flexbox to
  style footer bar and used list type for ul

- Syled latest-deals section using flexbox and image

dipbazz committed:
Updated style.css and index.html and created style768.css file

- created style768.css file for responsive breakpoint styles for 768px breakpoint.
- displayed top header and styled it on 768 screen
- remove menu icon and  menu list displayed on bottom header
- work on progress on banner news.

smcommits committed:
Update CSS

CSS
- made latest-news container responsive using grid layout
- used grid layout on funding section and made it responsive at 768px
- used grid layout on news-suggestion and made it responsive at 768px

dipbazz committed:
Update HTML and css, created new css for 1024px width

Changes on index.html
- moved footer from inside to outside of main element.
- grouped heading and side text on funding section for using flex

Created style1024.css
- made latest news, funding,latest deals to 4 grid column
- made news suggestion to 3 grid layout

smcommits committed:
Progress with HTML and CSS

HTML
- Added logo in banner-img large section

CSS
- Syled the fonts and hover effect on every section
- Fixed the height of banner image on mobile viewport
- Added gradient effect to the banner on largest breakpoint
- Placed the logo with position of absolute on banner image at largest
  breakpoint

dipbazz committed:
Added files, updated HTML and CSS after linter errors.

Added files
- app screenshot image

Updated HTML, CSS, and Readme:
- update readme file for the project.
- added class to links on HTML
- Chaged a selector to class selector on css
- fixed the linters error from HTML and CSS

smcommits committed:
Update README.md

dipbazz committed:
fixed linter errors of space on css files

dipbazz committed:
Merge branch 'responsive-design' of github.com:dipbazz/the-next-web-clone into responsive-design

umairarshadbutt commented on assets/css/style768.css:
```diff
@@ -0,0 +1,140 @@
+
+.news-article img {
+  width: 100%;
+  height: unset !important;
```
remove `!important`

umairarshadbutt commented on index.html:
```diff
@@ -0,0 +1,795 @@
+</head>
+<body>
+  <header>
+    <figure class="logo"><img src="assets/img/tnw.svg" alt=""></figure>
```
This is just an example but there are other images that are missing a description on the alternative text attribute.
Actual
`<figure class="logo"><img src="assets/img/tnw.svg" alt=""></figure>`
** Improved** +1 :
`<figure class="logo"><img src="assets/img/tnw.svg" alt="TNW Logo"></figure>`
Having an alternate text help SEO and Screen-readers

umairarshadbutt reviewed (changes requested):
# Status: REQUIRED CHANGES :warning: :construction: 

Hi @dipbazz and smcommits
Such a remarkable project you made :+1: ! Awesome README :100: !
We, however, do have a few changes to carry out before we proceed (see review comment).

##

 For this project, we have to make sure nothing is messed up. Kindly compare the screenshots below 👇 and work on the suggested changes.
- [x] Remove `!important `from css file. ([Article for support](https://speckyboy.com/good-bad-css-practices/)) ( [screenshot](https://user-images.githubusercontent.com/29689273/93404399-8997b500-f8a3-11ea-8f88-fb12b8f6ca80.png) from Pathwright)
- [x] Write a descriptive alternative text on `alt=""`, don't leave it blank. See inline comment.
##
- **When you have made the specified changes, please submit another code review. Thank you!**
##
**If you stick at any point or have any questions, let me know on slack at any time. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

smcommits committed:
Edit HTML and CSS

HTML
- Added a class 'news-image' to every image in the news-article section

CSS
- Removed the rule important from the images in news-article section

dipbazz committed:
included alt text on image

arslanbisharat reviewed (approved):
## Congratulations Approved 🎉
--------------------------------------------------------------------
:clap:
Hello Developer 👋
You are on 🔥
Thank you so much for making those changes and I am very happy to approve the project. You have done a great job. 👏
Please keep working hard and move forward with your curriculum.
Looking forward to seeing your next project. 😄
---------------------------------------------------------------------------------------------------------------
**To highlight**
- The cloned page is rendering as it should and meets all the project specifications.
-  CSS is clean.
-  The markup as well has no errors nor warnings to show. Good job with that.
-  The repo has followed all the guidelines there are to follow.
-  All the required tasks have been fulfilled in this project.
-  Congratulations on the splendid job👍
Now you can merge the branches and move forward with your curriculum. 🎆 

[OPTIONAL] If you wish to delete your currently working branch please update the Live demo link too so that it works properly. 

_Keep working hard._ 

I am very excited to see your next project.

**Happy Coding**:smiley:
------------------------------------------------------------------------------
**Regards,
Arslan Bisharat**
[linkedin](https://www.linkedin.com/in/arslanbisharat/) slack: @arslanbisharat
